### DISCUSSION TITLE
Attempting to getUri() returns error "instanceInfo" is undefined

### DISCUSSION THREAD

**Discussion by cbryant24** [typegoose/mongodb-memory-server#458] [Q&A] [answered]
## Versions

- NodeJS: v15.10.0
- mongodb-memory-server-*: 7.0.0-beta.26
- mongodb: 4.0.20
- mongoose: 5.11.19
- system: MacOS Big Sur Apple M1

package: mongo-memory-server 

## What is the Problem?

I Run `npm run test` and the test throws an error when invoking `await mongoServer.getUri()` stating the `"instanceInfo" 
is undefined`

<img width="861" alt="Screen Shot 2021-03-10 at 12 37 29 PM" src="https://user-images.githubusercontent.com/29960336/110694407-63d9d180-819d-11eb-8e4d-d4bf9725f580.png">

## Code Example

This repo can also be cloned and run to reproduce the problem if using an apple macbook with an m1 chip 
https://github.com/cbryant24/mongo-memory-server

```ts
import mongoose from "mongoose";
import { MongoMemoryServer } from "mongodb-memory-server";

// May require additional time for downloading MongoDB binaries
jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;

let mongoServer: any;

beforeAll(async () => {
  mongoServer = new MongoMemoryServer();
  const mongoUri = await mongoServer.getUri();
  await mongoose.connect(mongoUri, (err) => {
    if (err) console.error(err);
  });
});

afterAll(async () => {
  await mongoose.disconnect();
  await mongoServer.stop();
});
```

## Do you know *why* it happenes?

*no*

**Comment by hasezoey** [comment, marked as answer]
sorry, if you didnt read the changelog, you wouldnt know - there is no migration guide *yet*

`getUri` (on all classes) is now sync, not an promise anymore, an instance needs to be manually started with `.start` or directly with `.create`

**Reply by cbryant24** [reply]
I'm able to connect now aftering running this code for a `jest` setup

```javascript
beforeAll(async () => {
  mongoServer = new MongoMemoryServer();
  await mongoServer.start();

  const mongoUri = mongoServer.getUri();

  await mongoose.connect(mongoUri, (err) => {
    if (err) console.error(err);

    return;
  });
});
```
but it seems to hang and not run my tests. Anything I am missing?

<img width="1385" alt="Screen Shot 2021-03-12 at 12 56 14 PM" src="https://user-images.githubusercontent.com/29960336/110997530-6457a100-8332-11eb-9c13-3ea8c358a280.png">

**Reply by cbryant24** [reply]
Actually I was able to fix it by adding `useNewUrlParser` options to the `connect` method

```javascript
  await mongoose.connect(mongoUri, { useNewUrlParser: true }, (err) => {
    if (err) console.error(err);
  });
```
